arwen/elrondapi/bigFloatOps.go

```diff
@@ -0,0 +1,893 @@
+	metering.UseGas(gasToUse)
+
+	dest := managedType.GetBigFloatOrCreate(destinationHandle)
+	op1, op2, err := managedType.GetTwoBigFloat(op1Handle, op2Handle)
+	if arwen.WithFault(err, context, runtime.BigFloatAPIErrorShouldFailExecution()) {
+		return
+	}
+	if oneIsInfinity(op1, op2) {
+		_ = arwen.WithFault(arwen.ErrInfinityFloatOperation, context, runtime.BigFloatAPIErrorShouldFailExecution())
+		return
+	}
+	managedType.ConsumeGasForBigFloatCopy(dest, op1, op2)
+	dest.Add(op1, op2)
+}
+
+//export v1_4_bigFloatSub
+func v1_4_bigFloatSub(context unsafe.Pointer, destinationHandle, op1Handle, op2Handle int32) {
+	managedType := arwen.GetManagedTypesContext(context)
+	metering := arwen.GetMeteringContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+
+	gasToUse := metering.GasSchedule().BigFloatAPICost.BigFloatSub
+	metering.UseGas(gasToUse)
+
+	dest := managedType.GetBigFloatOrCreate(destinationHandle)
+	op1, op2, err := managedType.GetTwoBigFloat(op1Handle, op2Handle)
+	if arwen.WithFault(err, context, runtime.BigFloatAPIErrorShouldFailExecution()) {
+		return
+	}
+	if oneIsInfinity(op1, op2) {
+		_ = arwen.WithFault(arwen.ErrInfinityFloatOperation, context, runtime.BigFloatAPIErrorShouldFailExecution())
+		return
+	}
+	managedType.ConsumeGasForBigFloatCopy(dest, op1, op2)
+	dest.Sub(op1, op2)
+}
+
+//export v1_4_bigFloatMul
+func v1_4_bigFloatMul(context unsafe.Pointer, destinationHandle, op1Handle, op2Handle int32) {
+	managedType := arwen.GetManagedTypesContext(context)
+	metering := arwen.GetMeteringContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+
+	gasToUse := metering.GasSchedule().BigFloatAPICost.BigFloatMul
+	metering.UseGas(gasToUse)
+
+	dest := managedType.GetBigFloatOrCreate(destinationHandle)
+	op1, op2, err := managedType.GetTwoBigFloat(op1Handle, op2Handle)
+	if arwen.WithFault(err, context, runtime.BigFloatAPIErrorShouldFailExecution()) {
+		return
+	}
+	if oneIsInfinity(op1, op2) {
+		_ = arwen.WithFault(arwen.ErrInfinityFloatOperation, context, runtime.BigFloatAPIErrorShouldFailExecution())
+		return
+	}
+	managedType.ConsumeGasForBigFloatCopy(dest, op1, op2)
+	dest.Mul(op1, op2)
+}
+
+//export v1_4_bigFloatDiv
+func v1_4_bigFloatDiv(context unsafe.Pointer, destinationHandle, op1Handle, op2Handle int32) {
+	managedType := arwen.GetManagedTypesContext(context)
+	metering := arwen.GetMeteringContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+
+	gasToUse := metering.GasSchedule().BigFloatAPICost.BigFloatDiv
+	metering.UseGas(gasToUse)
+
+	dest := managedType.GetBigFloatOrCreate(destinationHandle)
+	op1, op2, err := managedType.GetTwoBigFloat(op1Handle, op2Handle)
+	if arwen.WithFault(err, context, runtime.BigFloatAPIErrorShouldFailExecution()) {
+		return
+	}
+	if oneIsInfinity(op1, op2) {
+		_ = arwen.WithFault(arwen.ErrInfinityFloatOperation, context, runtime.BigFloatAPIErrorShouldFailExecution())
+		return
+	}
+	if allAreEqualToZero(op1, op2) {
```

Comment:
Division by zero unhandled! Check op2 separately.

Reply:
if op2 is zero it won't panic. you'll get +- infinity . Then the next operation you will want to use that value, you will get execution failed as no operation with infinity is allowed. However, I will add extra checks everywhere so you cannot even initialize big floats  with +- infinity.